jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -828,12 +834,12 @@ protected void putHeaders(HttpServletResponse response, HttpContent content, lon
         {
             Response r = (Response)response;
             r.putHeaders(content, contentLength, _etags);
-            HttpFields f = r.getHttpFields();
-            if (_acceptRanges && !response.containsHeader(HttpHeader.ACCEPT_RANGES.asString()))
-                f.put(ACCEPT_RANGES);
+            HttpFields fields = r.getHttpFields();
+            if (_acceptRanges && !fields.contains(HttpHeader.ACCEPT_RANGES))
+                    fields.put(ACCEPT_RANGES);
```

Comment:
using `fields.put` adds the precomputed field directly to the headers without any further generation.